index.html

```diff
@@ -609,106 +609,211 @@
-    };
-  </pre>
-  <pre title="Save and restore game progress with another device"
-       class="example highlight">
-    // Watching for specific content; here only own game data.
-    navigator.nfc.watch(reader, { url: "*/mypath/mygame/*" });
+  <aside title="Push a text string to a peer device" class="example">
+    <p>
+      It is possible to restrict to which devices (tags or peers) data
+      should be pushed. Below we specify to only push to peers,
+      and thus, no data is pushed when the user taps a tag.
+    </p>
+    <pre class="highlight">
+navigator.nfc.push(
+    "Text meant for peers only", { target: "peer" }
+  ).then(() => {
+    console.log("Message pushed.");
+  }).catch((error) => {
+    console.log("Push failed :-( try again.");
+  });
+    </pre>
+  </aside>
 
-    function reader(message) {
-      console.log("Game state received from: " + message.url);
-      console.log("Stored state: " + message.data);
+  <aside title="Push a URL to either a tag or peer" class="example">
+    <p>
+      In order to push an NDEF record of URL type, simply use NFCMessage.
+    </p>
+    <pre class="highlight">
+navigator.nfc.push({
+    data: [{ recordType: "url", data: "/path/resource/" }];
+  }).then(() => {
+    console.log("Message pushed.");
+  }).catch((error) => {
+    console.log("Push failed :-( try again.");
+  });
+    </pre>
+  </aside>
 
-      // Now do some calculations and then update the state.
+  <aside title="Read and update data stored on tag" class="example">
+    <pre class="highlight">
+// Use default options for watch.
+navigator.nfc.watch((message) => {
+  console.log('NDEF message received from URL ' + message.url);
 
-      // Specify custom format, to control NDEF data layout.
-      var message = {
-        url: "https://company.com/path/game/update",  // app specific path
-        data: [{
-          recordType: "json",
-          mediaType: "application/json",
-          data: { level: 3, points: 4500, lives: 3 }
-        }]
-      };
+  if (message.data[0].recordType == 'empty') {
+    navigator.nfc.push({
+      url: "/mypath/myapp",  // path set by the application
+      data: [{
+        recordType: "url",
+        data: "Initial data"
+      }]
+    });
+  }
+  processMessage(message);
+}).then(() => {
+  console.log("Watch added.");
+}).catch((error) => {
+  console.log("Adding watch failed: " + error.name)
+});
+
+function processMessage(message) {
+  message.data.forEach((record) => {
+    if (record.recordType == "string") {
+      console.log('Data is string: ' + record.data);
+    } else if (record.recordType == 'json') {
+      processJSON(record.data);
+    } else if (record.recordType == 'url') {
+    console.log('Data is URL: ' + record.data);
+    } else if (record.recordType == 'opaque') {
+      processBinary(record.data);
+  });
+};
+
+function processBinary(data) {
+  console.log('Known (string) binary data: ');
+  console.log(String.fromCharCode.apply(null, new Uint16Array(data)));
```

Comment:
processBinary should check type and create proper view accordingly. Just creating Uint16 is misleading.
For example:
```javascript
if (record.type == "image/png") {
  somePlaceholderImg.src = URL.createObjectURL(new Blob(record.data, record.type));
  somePlaceholderImg.onload = function() {
    URL.revokeObjectURL(this.src);
  }
}
```